tests/integration/targets/test_zabbix_host/tasks/zabbix_host_tests.yml

```diff
@@ -1357,10 +1374,10 @@
             type: secret
       register: zabbix_host1
 
-    - name: expect to succeed and that things have not changed
+    - name: expect to succeed and that things have changed as it has secret value
       assert:
         that:
-          - "zabbix_host1 is not changed"
+          - "zabbix_host1 is changed"
```

Comment:
I think we may face streak of users that will complain about behavioral change. I am not particularly against thins change though

Reply:
Logic is simple: it's not correct to provide 'not changed' if you have no idea whether you are changing the value or not. Moreover if we report 'not changed' then we don't even call host.update() thus your changes to secret macro will never be applied.

Reply:
Yes, I agree. Just pointing out that there will be definitely someone to open an issue about :)